### PR TITLE
log-streaming lambda function: substitute epochTime for timestamp in cases it seems appropriate

### PR DESCRIPTION
**Pull request by risicle**
This is really a workaround for nginx, which is unable to log fractional
seconds in anything other than epoch time format, which the awslogs agent
isn't able to interpret as a timestamp, so we're using this lambda function
to perform a little post-processing and re-add the precision back into the
stream.

Already deployed on preview & seems to work ok there.

### CODE REVIEW

File: terraform/modules/log-streaming/src/main.js [hunk 1]
```diff
@@ -72,7 +72,20 @@ function transform(payload) {
 
         var source = buildSource(logEvent.message);
         source['instance_id'] = payload.logStream;
-        source['@timestamp'] = new Date(1 * logEvent.timestamp).toISOString();
+
+        if ("epochTime" in source && !isNaN(parseFloat(source.epochTime)) && isFinite(source.epochTime)
+            && Math.floor(Math.floor(source.epochTime) * 1000) === timestamp.getTime()) {
```
Comment (TheDoubleK) [inline terraform/modules/log-streaming/src/main.js]: I don't see that the outer `Math.floor` will ever do anything here, as the thing it's acting on is already guaranteed to be an integer.
e.g.
```
Math.floor(Math.floor(12345.09876) * 1000) 
= Math.floor(12345 * 1000)
= Math.floor(12345000)    // Always an integer in here so floor has no effect?
= 12345000
```
Or have I missed something?

Comment (risicle) [reply terraform/modules/log-streaming/src/main.js]: Belt and braces. Anyone coming along to this code doesn't even need to *think* about the possibility of it being non-integer if they see that `floor`. What was js's Number weirdness around the 53bit mark again? Doesn't matter.

Comment (risicle) [reply terraform/modules/log-streaming/src/main.js]: And also I'm just not risking something stupid going wrong in our log-critical code.

Comment (TheDoubleK) [reply terraform/modules/log-streaming/src/main.js]: It's confusing though (at least, it confused me and I had to spend time in a Javascript console convincing myself what was going on).  Is there really a possibility that the integer multiplication inside the brackets might give a non-integer? I don't see it.

Comment (risicle) [reply terraform/modules/log-streaming/src/main.js]: > Is there really a possibility that the integer multiplication inside the brackets might give a non-integer?

definitely not if you wrap it in `floor`

**Pull request merged**